app/src/main/java/com/asfoundation/wallet/base/SingleStateFragment.kt

```diff
@@ -0,0 +1,30 @@
+package com.asfoundation.wallet.base
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleCoroutineScope
+import androidx.lifecycle.flowWithLifecycle
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+interface SingleStateFragment<S : ViewState, E : SideEffect> {
+
+  fun onStateChanged(state: S)
+
+  fun onSideEffect(sideEffect: E)
+
+  fun BaseViewModel<S, E>.collectStateAndEvents(lifecycle: Lifecycle,
```

Comment:
It's simply an helper function that represents the usage of 95% of the cases. The function itself is to be used inside the fragment, so the only alternative to this would be to actually create a BaseFragment that would contain this. This is discouraged because we lose flexibility when we start creating hierarchies / abstract classes (e.g. what about the dagger code? what about DialogFragments? etc.). The other issue is that there are cases (the 5%) where you may want to subscribe to specific property fields (`stateFlow .map { state -> state.property }.distinctUntilChanged()`) and you may not want to use this helper function / interface.

In my project I actually am not even using any interface nor helper function, so we can discuss if this is even needed or not, or improvements to it.